### PR TITLE
Add google.com

### PR DESCRIPTION
**Pull request by gobinathm**
- Include password change URL For google.com
- Include password rule for google.com

### CODE REVIEW

**Comment by karbassi** [comment]
Shouldn't we add all the `google.tld` domains to websites-with-shared-credential-backends.json as well?

**Comment by gobinathm** [comment]
> Shouldn't we add all the `google.tld` domains to websites-with-shared-credential-backends.json as well?

Have added some of the Google shared domain. Anyone else looking at this PR can add more

**Comment by igor-makarov** [comment]
How about `google.co.il`?

**Comment by igor-makarov** [comment]
Wow, they sure have a lot of TLDs...

**Comment by mnoorenberghe** [comment]
Are credentials actually entered on the Google ccTLDs? I thought all logins for Google itself go through accounts.google.com or myaccount.google.com

**Comment by gobinathm** [comment]
> Are credentials actually entered on the Google ccTLDs? I thought all logins for Google itself go through accounts.google.com or myaccount.google.com

To my knowledge this differences from the location.

**Comment by igor-makarov** [comment]
I've tried `google.co.il` and the login is still at `accounts.google.com`.

**Comment by gobinathm** [comment]
> I've tried `google.co.il` and the login is still at `accounts.google.com`.

True Indeed. Valided 50% of the domain and every one redirected to accounts.google.com so removed them from shared credentials

File: quirks/change-password-URLs.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "ea.com": "https://myaccount.ea.com/cp-ui/security/index",
     "fnac.com": "https://secure.fnac.com/account/update-password",
     "geocaching.com": "https://www.geocaching.com/account/settings/changepassword",
+    "google.com": "https://myaccount.google.com/signinoptions/password",
```
[the remark this answers is not in the document]
Comment (rmondello) [reply quirks/change-password-URLs.json:15]: Google supports the well-known URL for changing password at accounts.google.com/.well-known/change-password. Until Google actually implements it for google.com itself, I think using that URL would be more principled.

File: quirks/password-rules.json [hunk 2]
```diff
@@ -99,6 +99,9 @@
     "fc2.com": {
         "password-rules": "minlength: 8; maxlength: 16;"
     },
+    "google.com": {
+        "password-rules": "minlength: 8; allowed: lower,upper,special; required: [@#$%^&+=_]; required: digit; allowed: upper, lower;"
```
[the remark this answers is not in the document]
Comment (gobinathm) [reply quirks/password-rules.json:109]: validating all the rules of Google. will update this request.

File: quirks/change-password-URLs.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (HarshPatel270698) [reply quirks/change-password-URLs.json:15]: > 
> 
> Google supports the well-known URL for changing password at accounts.google.com/.well-known/change-password. Until Google actually implements it for google.com itself, I think using that URL would be more principled.

But https://myaccount.google.com/signinoptions/password link is appearing at the top in Google search
Keywords used by me in Google Search were "reset google password".

[the remark this answers is not in the document]
Comment (rmondello) [reply quirks/change-password-URLs.json:15]: And if that URL changes, the Well-Known URL should hopefully redirect to the new location!

**Comment by rmondello** [comment]
Hi @gobinathm! I’m still interested in getting this merged. Do you wanna clean it up so it merges and address any outstanding feedback?

**Comment by jumoru** [comment]
Google outlines their password requirements on this site:
https://support.google.com/accounts/answer/32040?hl=en

image

Given their last bullet point we probably have to leave out spaces, so something like this would probably work:

`minlength: 8; allowed: lower, upper, digit, [-!\"#$%&'()*+,./:;<=>?@[^_`{|}~]];

**Comment by rmondello** [comment]
Closing this due to inactivity. Please feel free to reopen, or someone else can take a shot at google.com.

**Pull request closed without merging**